File: common/library-framework/src/main/java/com/donews/core/utils/MetaDataUtils.java
package com.donews.core.utils;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.text.TextUtils;

/**
 * App MetaData相关工具类
 *
 * @author XuShuai
 * @version v1.0
 * @date 2021/4/7
 */
public class MetaDataUtils {

    public static String getMetaData(Context context, String key) {
        if (context == null || TextUtils.isEmpty(key)) {
            return null;
        }
        String value = null;
        try {
            PackageManager packageManager = context.getPackageManager();
            if (packageManager != null) {
                ApplicationInfo applicationInfo = packageManager.getApplicationInfo(context.getPackageName(),
                        PackageManager.GET_META_DATA);
                if (applicationInfo != null && applicationInfo.metaData != null) {
                    value = applicationInfo.metaData.getString(key);
                }
            }
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return value;
    }

}

File: common/library-core/src/main/java/com/adair/core2/utils/MetaDataUtils.kt
package com.adair.core2.utils

import android.content.Context
import android.content.pm.PackageManager

/**
 * App 读取MetaData相关工具类
 *
 * @author XuShuai
 * @version v1.0
 * @date 2022/2/24 10:21
 */
class MetaDataUtils {

    companion object {

        /**
         * 读取 配置的MetaData 指定key的值
         * @param context Context 上下文对象
         * @param key String metaData的 name
         * @return String? 值，可能为null
         */
        @JvmStatic
        fun getMetaData(context: Context, key: String): String? {
            if (key.isBlank()) {
                return null
            }

            var value: String? = null
            try {
                val packageManager = context.packageManager
                val applicationInfo =
                    packageManager.getApplicationInfo(context.packageName, PackageManager.GET_META_DATA)
                if (applicationInfo.metaData != null) {
                    value = applicationInfo.metaData.getString(key)
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return value
        }
    }
}